### PR TITLE
Golang version to 1.18 and k8s client to 1.24

### PR DESCRIPTION
**Pull request by aminesnow**
### Description

Update to use Go 1.18 and k8s client to 0.24.3.

/cc @periklis 
/assign @periklis 

### Links

- JIRA: https://issues.redhat.com/browse/LOG-2834

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ### This is a generated file from Dockerfile.in ###
-#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.17)
-FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.10 AS builder
+#@follow_tag(registry-proxy.engineering.redhat.com/rh-osbs/openshift-golang-builder:rhel_8_golang_1.18)
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.18-openshift-4.10 AS builder
```
Comment (periklis) [inline Dockerfile:3]: We can use the `4.11` builder too.

File: go.mod [hunk 2]
```diff
@@ -13,10 +13,10 @@ require (
...
 	github.com/prometheus/client_golang v1.12.1
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.23.5
-	k8s.io/apimachinery v0.23.5
-	k8s.io/client-go v0.23.5
-	k8s.io/utils v0.0.0-20211116205334-6203023598ed
+	k8s.io/api v0.24.0
+	k8s.io/apimachinery v0.24.0
+	k8s.io/client-go v0.24.0
+	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
```
Comment (periklis) [inline go.mod:19]: Why not the latest minor release for 1.24? I believe the latest is `1.24.3`

File: internal/kibana/trustedcabundle.go [hunk 3]
```diff
@@ -75,7 +75,7 @@ func calcTrustedCAHashValue(configMap *corev1.ConfigMap) (string, error) {
 	if !ok {
 		return "", kverrors.New("expected key does not exist in configmap",
 			"key", constants.TrustedCABundleKey,
-			"cluster", configMap.ClusterName,
+			"cluster", configMap.ZZZ_DeprecatedClusterName, // This is deprecated. We either stop using it or find another way.
```
Comment (periklis) [inline internal/kibana/trustedcabundle.go:78]: We should stop using it.

**Review by periklis — COMMENTED** [review]
Looks good so far. Remember you need to update also the upstream CI to get the builders/tests pass here https://github.com/openshift/release/blob/master/ci-operator/config/openshift/elasticsearch-operator/openshift-elasticsearch-operator-master.yaml#L1-L72:
- Base images should go up to `4.11`
- RHEL8 Golang builder should be 1.18 from 4.11

File: go.mod (hunk 2 above)
Comment (aminesnow) [reply go.mod]: sure

**Review by periklis — COMMENTED** [review]
Waiting CI verdict

**Comment by Red-GV** [comment]
Looks good. @aminesnow, please squash commits.

**Comment by aminesnow** [comment]
> Looks good. @aminesnow, please squash commits.

Done.

**Pull request merged**